pyprof/nvtx/nvmarker.py

```diff
@@ -267,22 +272,105 @@ def modMarker(mod, fn_name, args):
+                                'shape': str(item.shape)
+                            }
+                        )
+                        if isinstance(item, list) or isinstance(item, tuple):
+                            for item2 in item:
+                                if isinstance(item2, torch.Tensor):
+                                    input_tensors.append(
+                                        {
+                                            'ptr': item2.data_ptr(),
+                                            'grad_fn': str(item2.grad_fn),
+                                            'grad_exists': (item2.grad is not None),
+                                            'shape': str(item2.shape)
+                                        }
+                                    )
+        for input_id, _ in enumerate(input_tensors):
+            input_ptr = input_tensors[input_id]['ptr']
+            if input_ptr in op_to_out_tensor_map:
+                input_callid_info = op_to_out_tensor_map[input_ptr]
+                if input_callid_info not in input_callid_list:
+                    input_callid_list.append(input_callid_info)
+
+    def capture_outputs(call_id, result):
+        output_tensors = []
+        if isinstance(result, torch.Tensor):
+            output_tensors.append(
+                {
+                    'ptr': result.data_ptr(),
+                    'grad_fn': str(result.grad_fn),
+                    'grad_exists': (result.grad is not None),
+                    'shape': str(result.shape)
+                }
+            )
+        elif isinstance(result, list) or isinstance(result, tuple):
+            for item in result:
+                if isinstance(item, torch.Tensor):
+                    output_tensors.append(
+                        {
+                            'ptr': item.data_ptr(),
+                            'grad_fn': str(item.grad_fn),
+                            'grad_exists': (item.grad is not None),
+                            'shape': str(item.shape)
+                        }
+                    )
+        for out_port, _ in enumerate(output_tensors):
+            output_ptr = output_tensors[out_port]['ptr']
+            op_to_out_tensor_map[output_ptr] = f"{call_id}"
 
     def wrapper_func(*args, **kwargs):
         global wrappers_enabled
+        global call_id
+        global op_to_out_tensor_map
+        global patch_list
+        global call_id_to_op_map
+
+        patch_list.append(call_id)
+        input_callid_list = []
 
-        if (wrappers_enabled):
+        if config.capture_input_ops:
+            capture_inputs(input_callid_list, *args)
+
+        if wrappers_enabled:
             # Push trace marker
-            nvtx.range_push(traceMarker(fn_name))
+            traceMarker_str = traceMarker(fn_name)
+            nvtx.range_push(traceMarker_str)
+            traceMarker_str = traceMarker_str.replace("\'", "\"")
+            traceMarker_dict = json.loads(traceMarker_str)
```

Comment:
Move this down into the code that consumes it. Otherwise it is adding overhead when not enabled.